checkov/terraform/base_runner.py

```diff
@@ -106,10 +107,14 @@ def load_external_checks(self, external_checks_dir: list[str] | None) -> None:
                 resource_registry.load_external_checks(directory)
                 self.graph_registry.load_external_checks(directory)
 
-    def get_connected_node(self, entity: dict[str, Any], root_folder: str) -> Optional[Dict[str, Any]]:
-        connected_entity = entity.get("connected_node")
-        if not connected_entity:
+    def _get_connected_node_data(self, connected_nodes_per_resource_types: dict[tuple[str], Any], root_folder: str,
+                                 connected_resource_types: list[tuple[str]]) -> Optional[Dict[str, Any]]:
+        if not connected_resource_types or not connected_nodes_per_resource_types:
             return None
+        existing_tuple = next((item for item in connected_resource_types if item in connected_nodes_per_resource_types), None)
```

Comment:
can you extract this one-liner to a function to make it more readable? 
also `existing_tuple` is an unclear name